File: Selenium/Activities/Activity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains

with webdriver.Firefox() as ff:

    ff.get("https://v1.training-support.net/selenium/drag-drop")
    print("title is : ", ff.title)

    act = ActionChains(ff)
    ball = ff.find_element(By.XPATH, "//div/img[@id='draggable']")
    zone1 = ff.find_element(By.XPATH, "//div[@id='droppable']")
    zone2 = ff.find_element(By.XPATH, "//div[@id='dropzone2']")

    act.click_and_hold(ball).move_to_element(zone1).pause(3).release().perform()

    if zone1.text.find("Dropped!"):
        print("Ball dropped in zone 1")
    

    act.drag_and_drop(ball,zone2).pause(3).perform()

    if zone2.text.find("Dropped!"):
        print("Ball dropped in zone 2")

File: Selenium/Activities/Activity7.java
package activities;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity7 {

	public static void main(String[] args) {

		WebDriver driver = new FirefoxDriver();
		driver.get("https://v1.training-support.net/selenium/drag-drop");

		System.out.println("The Title of the page is  : " + driver.getTitle());

		Actions act = new Actions(driver);

		WebElement ball = driver.findElement(By.xpath("//div/img[@id='draggable']"));
		WebElement zone1 = driver.findElement(By.xpath("//div[@id='droppable']"));
		WebElement zone2 = driver.findElement(By.xpath("//div[@id='dropzone2']"));

		act.clickAndHold(ball).moveToElement(zone1).pause(5000).release().build().perform();

		String dropzone1Verify = zone1.findElement(By.tagName("p")).getText();
		if (dropzone1Verify.equals("Dropped!")) {
			System.out.println("Ball was dropped in dropzone 1");
		}

		act.clickAndHold(ball).moveToElement(zone2).pause(5000).release().build().perform();

		String verify2 = zone1.findElement(By.tagName("p")).getText();
		if (verify2.equals("Dropped!")) {
			System.out.println("Ball was dropped in dropzone 2");
		}

		driver.quit();

	}

}
